pkg/asset/installconfig/azure/metadata.go

```diff
@@ -19,12 +19,24 @@ type Metadata struct {
 	// CloudName indicates the Azure cloud environment (e.g. public, gov't).
 	CloudName typesazure.CloudEnvironment `json:"cloudName,omitempty"`
 
+	// Credentials hold prepopulated Azure credentials.
+	// At the moment the installer doesn't use it and reads credentials
+	// from the file system, but external consumers of the package can
+	// provide credentials. This is useful when we run the installer
+	// as a service (Azure Red Hat OpenShift, for example): in this case
+	// we do not want to rely on the filesystem or user input as we
+	// serve multiple users with different credentials via a web server.
+	Credentials *Credentials `json:"credentials,omitempty"`
+
 	mutex sync.Mutex
 }
 
 // NewMetadata initializes a new Metadata object.
-func NewMetadata(cloudName typesazure.CloudEnvironment) *Metadata {
-	return &Metadata{CloudName: cloudName}
+func NewMetadata(cloudName typesazure.CloudEnvironment, credentials *Credentials) *Metadata {
```

Comment:
I did not go for this initially because I was a bit afraid that it might look redundant when someone looks here without taking external usage into account. But I hope the code comment around credentials and the fact that this function is exported will deter future us from refactoring and removing `NewMetadataWithCredentials` :)

Reply:
I share your concern. However, I would be just as concerned about someone removing the `credentials` field.